### PR TITLE
Add initial product page template behind feature flag

### PR DESCRIPTION
**Pull request by dhmacs**
This adds a product page template stub behind a feature flag, as we don't want to release this before it's ready. 
I'll be adding one by one the pieces that I've worked on last month, this is just to start developing with feature flags.

This implementation uses environment variables to control flags.

## QA

1. Open [prod-like Vercel preview](https://react-commerce-prod-git-add-product-page-template-ifixit.vercel.app/Products/test) (flag not enabled)
2. Verify that `/Products/test` returns a 404
3. Visit [staging Vercel preview](https://react-commerce-git-add-product-page-template-ifixit.vercel.app/Products/test) (flag is enabled)
4. Verify that you can see a product page stub at `/Products/test`

### CODE REVIEW

**Comment by jordycosta** [comment]
QA 🟢 

The product page stub loads for when the flag is enabled; and otherwise, returns a 404 page. ✔️ 

---

### Testing Procedure:
1. Navigate to the flag-disabled Vercel preview: https://react-commerce-prod-git-add-product-page-template-ifixit.vercel.app/Products/test
> Check that this returns a 404 page

2. Navigate to the flag-enabled Vercel preview: https://react-commerce-git-add-product-page-template-ifixit.vercel.app/Products/test

> We should see a product stub page

---

### Screenshots:



product-page-stub

File: frontend/config/flags.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export const flags = {
+   PRODUCT_PAGE_ENABLED:
+      process.env.NEXT_PUBLIC_FLAG__PRODUCT_PAGE_ENABLED === 'true',
```
Comment (ardelato) [inline frontend/config/flags.ts:3]: Is this the same as saying?

```
	PRODUCT_PAGE_ENABLED: process.env.NEXT_PUBLIC_FLAG__PRODUCT_PAGE_ENABLED ?? false
```

Comment (ardelato) [reply frontend/config/flags.ts:3]: Ahh scratch that, I see that `process.env.NEXT_PUBLIC_FLAG__PRODUCT_PAGE_ENABLED` can be a string so `'false'` would return true.

**Review by ardelato — APPROVED** [review]
CR 🦖

**Pull request merged**